assets/webhook/validating_webhook_configuration.yaml

```diff
@@ -0,0 +1,28 @@
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingWebhookConfiguration
```

Comment:
somewhat related to @jkhelil 's ordering question, this config is missing the namespace selector discussed in https://github.com/openshift/enhancements/blob/master/enhancements/cluster-scope-secret-volumes/shared-resource-validation.md#api-extensions

You also need to check all the apps/v1 types that turn into pods, like deployments, daemonsets, etc.

Also, adding the `csi.sharedresource.openshift.io/skip-validation: true` label to the csi driver deployment should exempt it from the filter getting called for it